### PR TITLE
Add overlay example

### LINKED ISSUE #200: Add overlay example

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #200.

cc @sdroege @EPashkin

### CODE REVIEW

File: src/bin/overlay.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+use std::env::args;
+
+// Basic CSS: we change background color, we set font color to black and we set it as bold.
+const STYLE: &'static str = "
+#countdown-label {
+    background-color: rgba(192, 192, 192, 0.8);
+    color: black;
+    font-weight: bold;
+}";
```
Comment (sdroege) [inline src/bin/overlay.rs]: I know this CSS ;)

File: src/bin/overlay.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        Inhibit(false)
+    });
+
+    // The overlay container.
+    let overlay = gtk::Overlay::new();
+
+    // The overlay label.
+    let overlay_text = gtk::Label::new("0");
+    // We need to name it in order to apply CSS on it.
+    gtk::WidgetExt::set_name(&overlay_text, "countdown-label");
```
Comment (sdroege) [inline src/bin/overlay.rs]: Why is it called countdown if we don't do one?

File: src/bin/overlay.rs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    overlay_text.set_halign(gtk::Align::End);
+    overlay_text.set_valign(gtk::Align::Start);
+
```
Comment (sdroege) [inline src/bin/overlay.rs]: Why not centered?

File: src/bin/overlay.rs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        }
+    }};
+    ($x:ident) => {
+        upgrade_weak!($x, ())
+    };
+}
+
+fn button_clicked(button: &gtk::Button, overlay_text_weak: &glib::object::WeakRef<gtk::Label>) {
+    let overlay_text = upgrade_weak!(overlay_text_weak);
+    overlay_text.set_text(&button.get_label().unwrap_or_else(|| String::new()));
```
Comment (sdroege) [inline src/bin/overlay.rs]: The unwrap can never fail, so expect :)

File: src/bin/overlay.rs (hunk 3 above)
Comment (EPashkin) [reply src/bin/overlay.rs]: IMHO current `set_halign(gtk::Align::End);` is better that centered.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks

File: src/bin/overlay.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/bin/overlay.rs]: Where could it come from?! :D

File: src/bin/overlay.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/bin/overlay.rs]: Ah true. I'll rename it.

File: src/bin/overlay.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply src/bin/overlay.rs]: Sounds better for the current case.

File: src/bin/overlay.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply src/bin/overlay.rs]: Yes sir!

File: src/bin/overlay.rs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            .expect("Failed to load CSS");
+        gtk::StyleContext::add_provider_for_screen(
+            &gdk::Screen::get_default().expect("Error initializing gtk css provider."),
+            &provider,
+            gtk::STYLE_PROVIDER_PRIORITY_APPLICATION,
+        );
+
+        // We build the application UI.
+        build_ui(app);
+    });
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/bin/overlay.rs]: Ok!

**Comment by sdroege** [comment]
:+1:

**Comment by EPashkin** [comment]
👍

**Pull request merged**